crates/xlinectl/src/command/lock.rs

```diff
@@ -23,20 +26,32 @@ pub(crate) fn build_request(matches: &ArgMatches) -> LockRequest {
 
 /// Execute the command
 pub(crate) async fn execute(client: &mut Client, matches: &ArgMatches) -> Result<()> {
-    let req = build_request(matches);
+    let lease_client = client.lease_client();
+    let lease_resp = lease_client
+        .grant(LeaseGrantRequest::new(DEFAULT_SESSION_TTL))
+        .await?;
+    let lock_lease_id = lease_resp.id;
+    let req = build_request(matches).with_lease(lock_lease_id);
+    let lock_resp = client.lock_client().lock(req).await?;
 
-    let resp = client.lock_client().lock(req).await?;
+    lock_resp.print();
 
-    resp.print();
+    let (keeper, stream) = client
+        .lease_client()
+        .keep_alive(LeaseKeepAliveRequest::new(lock_lease_id))
+        .await?;
```

Comment:
We could  wrap these code in to a new "session" type in `xline-client`, instead of letting the user to keep alive the lock manually.
FYI: https://github.com/etcd-io/etcd/blob/main/client/v3/concurrency/session.go

Reply:
I have the same idea. This should be a refactor, I'll leave an issue to track it. #684